app/jobs/cleanup_files_job.rb

```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+class CleanupFilesJob < ApplicationJob
+  delegate :query_service, to: :metadata_adapter
+
+  def perform(file_identifiers:)
+    file_identifiers.each do |file_identifier|
+      storage_adapter = Valkyrie::StorageAdapter.adapter_for(id: file_identifier)
+      # Delete the entire directory to remove unzipped derivatives like display vectors
+      id = File.dirname(file_identifier.to_s)
```

Comment:
This works for now, but feels like it relies on the delete implementation of the disk storage adapter. I'm not sure what the alternative is...maybe something in Valkyrie?

Reply:
I had the same thought, but I couldn't think of a method for handling the more general case. I think in the future we'll load vector data into Postgres/PostGIS or use a single file format (like GeoPackage) and this won't be an issue.